### ISSUE TITLE
[Bug] Big Increase in Network Traffic after upgradeing to 2.*

### ISSUE THREAD

**Issue by Omar-Bishtawi** [kubecost/features-bugs#64]
### Kubecost Version

2.2.0

### Kubernetes Version

1.29

### Kubernetes Platform

EKS

### Description

after upgrading kubecost from 1.108.1 to 2.2.0 , there was a big increase in the outgoing network traffic.
- before the upgrade around 500Mb each day 
- after the upgrade around 250GB each day
<img width="1478" alt="image" src="https://github.com/kubecost/cost-analyzer-helm-chart/assets/91823055/a0097532-7b0c-483c-b910-57c3d7cdf85f">

### Steps to reproduce

1. upgrade kubecost from <2 version to 2>= version 
```bash
helm upgrade -i kubecost oci://public.ecr.aws/kubecost/cost-analyzer \
    --version 2.2.0 \
    --namespace kubecost --create-namespace \
    --set networkCosts.enabled=true \
    --set prometheus.server.resources.requests.memory=16Gi \
    --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=<IAM_ROLE> \
    --set kubecostProductConfigs.athenaProjectID=<PROJECT_ID> \
    --set kubecostProductConfigs.athenaBucketName=<CUR_S3_LOCATION>
    --set kubecostProductConfigs.athenaRegion=us-east-1 \
    --set kubecostProductConfigs.athenaDatabase=athenacurcfn_main_report \
    --set kubecostProductConfigs.athenaTable=main_report \
    --set kubecostProductConfigs.athenaWorkgroup=AthenaV2 \
    --set 'kubecostModel.extraEnv[0].name=LOG_FORMAT,kubecostModel.extraEnv[0].value=JSON' \
    -f https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-eks-cost-monitoring.yaml
```
2. monitor incoming network traffic size  from `athena.us-east-1.amazonaws.com` domain before and after the upgrade. the increase in traffic is coming from the `cloud-cost` container. 

### Expected behavior

not sure if this behavior is expected or not, if it's expected I think it's worth mentioning an expect in Network traffic cost if you enable the feature that is causing the increase in network traffic. 

### Impact

big increase in network traffic cost depending on the scale of your account..

### Troubleshooting

- [X] I have read and followed the [issue guidelines](https://github.com/kubecost/features-bugs/blob/main/ISSUE_GUIDELINES.md) and this is a bug impacting only the Kubecost application.
- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by Sean-Holcomb** [comment]
Can you share some of your logs for the cloud-cost container?
additionally can you share the output of `/cloudCost/status` endpoint?

**Comment by Sean-Holcomb** [comment]
Additionally, if you have access to Athena, can you see what the output bytes are for the query being made by Kubecost is. It should be something along the lines of 
```
SELECT DATE_TRUNC('day',line_item_usage_start_date) as usage_date, line_item_resource_id, bill_payer_account_id, line_item_usage_account_id, line_item_product_code, line_item_usage_type, (line_item_product_code = 'AmazonEKS' OR resource_tags_aws_eks_cluster_name <> '' OR resource_tags_user_eks_cluster_name <> '' OR...
```

This would be under "Recent Queries" in the Athena console

**Comment by Omar-Bishtawi** [comment]
Hi @Sean-Holcomb 
attached all log lines for current running pod of kubecost ( was running for almost 1 day ) 
[extract-2024-04-03T08 02 55.882Z.csv](https://github.com/kubecost/features-bugs/files/14848858/extract-2024-04-03T08.02.55.882Z.csv)
the only ERR log I see is this one and it only happned once at pod startup everything else looks normal. 
> error reading files in directory /opt/opencost/plugin/config: open /opt/opencost/plugin/config: no such file or directory

for the `/cloudCost/status` here is the outut:
```json
{"code":200,"data":[{"key":"095326208734/s3://aws-athena-query-results-095326208734-us-east-1","source":"multicloud","provider":"AWS","active":true,"valid":true,"lastRun":"2024-04-03T08:12:56.722563181Z","nextRun":"2024-04-03T14:12:56.722563181Z","RefreshRate":"6h0m0s","created":"2024-04-03T08:12:56.722563181Z","runs":0,"coverage":"[2024-04-04T00:00:00+0000, 2024-04-04T00:00:00+0000)","connectionStatus":"No Connection","config":{"bucket":"s3://aws-athena-query-results-095326208734-us-east-1","region":"us-east-1","database":"athenacurcfn_main_report","catalog":"","table":"main_report","workgroup":"AthenaV2","account":"095326208734","authorizer":{"authorizerType":"AWSServiceAccount"}}},{"key":"095326208734/s3://aws-athena-query-results-095326208734-us-east-1","source":"configfile","provider":"AWS","active":false,"valid":true,"lastRun":"2024-04-03T08:12:56.722563181Z","nextRun":"2024-04-03T14:12:56.722563181Z","RefreshRate":"6h0m0s","created":"2024-04-03T08:12:56.722563181Z","runs":0,"coverage":"[2024-04-04T00:00:00+0000, 2024-04-04T00:00:00+0000)","connectionStatus":"No Connection","config":{"bucket":"s3://aws-athena-query-results-095326208734-us-east-1","region":"us-east-1","database":"athenacurcfn_main_report","catalog":"","table":"main_report","workgroup":"AthenaV2","account":"095326208734","authorizer":{"authorizerType":"AWSServiceAccount"}}}]}
```

but please note that the page took along time to generate this json field ( almost 2 min) .


for Athena queiry size here you can see the output size for last 17 queries :
<img width="877" alt="image" src="https://github.com/kubecost/features-bugs/assets/91823055/8087f18e-7453-4126-b821-c06dc2000f81">

**Comment by Sean-Holcomb** [comment]
Awesome, those logs have a great clue of what the issue is, looking into the a fix for it. If you want a fix in the short term, you may want to try creating a secret for your athena configuration instead of setting those helm values.
https://docs.kubecost.com/install-and-configure/install/cloud-integration/multi-cloud
You should also be sure to blank out the helm values

**Comment by Sean-Holcomb** [comment]
Actually before you do that can you share the output of `kubectl get secrets`

**Comment by Sean-Holcomb** [comment]
There seems to be two problems and I think both are related to your PV. One is that the state of the configs in not properly saving causing the container to treat existing configurations like new configurations, which resets the process. The second problem is that the cloud cost from the runs that are happening are not saving. On start up the integration checks for historic  data and builds it if it is not present. Because data is not saving it tries to rebuild the entire history of CloudCost every time it restarts.

**Comment by Omar-Bishtawi** [comment]
here is the active secrets on kubecost.
```bash
❯ k get secrets -n kubecost
NAME                              TYPE                 DATA   AGE
cloud-integration                 Opaque               1      43h
kubecost-plugin-secret            Opaque               0      43h
sh.helm.release.v1.kubecost.v11   helm.sh/release.v1   1      80d
sh.helm.release.v1.kubecost.v12   helm.sh/release.v1   1      80d
sh.helm.release.v1.kubecost.v13   helm.sh/release.v1   1      80d
sh.helm.release.v1.kubecost.v14   helm.sh/release.v1   1      3d23h
sh.helm.release.v1.kubecost.v15   helm.sh/release.v1   1      44h
sh.helm.release.v1.kubecost.v16   helm.sh/release.v1   1      44h
sh.helm.release.v1.kubecost.v17   helm.sh/release.v1   1      43h
sh.helm.release.v1.kubecost.v18   helm.sh/release.v1   1      43h
sh.helm.release.v1.kubecost.v19   helm.sh/release.v1   1      43h
sh.helm.release.v1.kubecost.v20   helm.sh/release.v1   1      43h
```
and I confirmed that content of cloud-integration is correct and they are same as in the [document]( https://docs.kubecost.com/install-and-configure/install/cloud-integration/multi-cloud) you sent are the same as the helm chart values.


also the secret is mounted correctly on the cloudcost container . 

note that this secret was created by the helm chart I didn't create it. so no actions are needed if understood your recommendation correctly.

**Comment by Sean-Holcomb** [comment]
Yes no action needed there.

I now believe there is an issue with the PV on your deployment, can you exec into your cloud-cost container and check the size of the directory mounted at `/var/configs/` and compare that against the size of the pv which attached to the pvc `kubecost-cost-analyzer`

**Comment by Sean-Holcomb** [comment]
@Omar-Bishtawi so I have dug a bit deeper today, and found that my original advice on creating the secret was actually correct for a short term fix, the reason is difficult to explain here and I am working on a longer term fix but you should manually create a secret with the same content as the one that was created by helm and run your helm command as so
```
helm upgrade -i kubecost oci://public.ecr.aws/kubecost/cost-analyzer \
    --version 2.2.0 \
    --namespace kubecost --create-namespace \
    --set networkCosts.enabled=true \
    --set prometheus.server.resources.requests.memory=16Gi \
    --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=<IAM_ROLE> \
    --set kubecostProductConfigs.athenaProjectID="" \
    --set kubecostProductConfigs.athenaBucketName="" \
    --set kubecostProductConfigs.athenaRegion="" \
    --set kubecostProductConfigs.athenaDatabase="" \
    --set kubecostProductConfigs.athenaTable="" \
    --set kubecostProductConfigs.athenaWorkgroup="" \
    --set kubecostProductConfigs.cloudIntegrationSecret=<MANUALLY_CREATED_SECRET_NAME> \
    --set 'kubecostModel.extraEnv[0].name=LOG_FORMAT,kubecostModel.extraEnv[0].value=JSON' \
    -f https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-eks-cost-monitoring.yaml
```

Blanking out these values and creating the secret manually should at least partially address the issue here. There may still be  problem with your pv saving cloud-cost, but you should no longer have issues with the massive network spike

**Comment by Omar-Bishtawi** [comment]
Thank you for the tip @Sean-Holcomb . 
after creating the secret manually and setting the value in `cloudIntegrationSecret` everything seems to work fine now.

**Comment by Sean-Holcomb** [comment]
Great, working on a fix for that configuration method

**Comment by dwbrown2** [comment]
@Sean-Holcomb were you able to drop a PR for this one? Good to mark as closed or still working?

**Comment by Sean-Holcomb** [comment]
Yes this can be closed, the fix will be in v2.2.2
